worldedit-bukkit/src/main/java/com/boydti/fawe/bukkit/adapter/mc1_16_5/TuinityRelighter_1_16_5.java

```diff
@@ -0,0 +1,151 @@
+
+    private final IQueueExtent<IQueueChunk> queue;
+
+    private final ReentrantLock lightLock = new ReentrantLock();
+    private final LongList chunks = new LongArrayList();
+
+    static {
+        MethodHandle tmp = null;
+        try {
+            Method relightMethod = LightEngineThreaded.class.getMethod(
+                    "relight",
+                    Set.class,
+                    Consumer.class,
+                    IntConsumer.class
+            );
+            tmp = MethodHandles.lookup().unreflect(relightMethod);
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            LOGGER.error("Failed to locate relight method in LightEngineThreaded on Tuinity. " +
+                    "Is everything up to date?", e);
+        }
+        relight = tmp;
+    }
+
+    public TuinityRelighter_1_16_5(WorldServer world, IQueueExtent<IQueueChunk> queue) {
+        this.queue = queue;
+        this.world = world;
+    }
+
+    @Override
+    public boolean addChunk(int cx, int cz, byte[] skipReason, int bitmask) {
+        lightLock.lock();
+        try {
+            chunks.add(ChunkCoordIntPair.pair(cx, cz));
+        } finally {
+            lightLock.unlock();
+        }
+        return true;
+    }
+
+    @Override
+    public void addLightUpdate(int x, int y, int z) {
+
+    }
+
+    @Override
+    public void fixLightingSafe(boolean sky) {
+        Set<ChunkCoordIntPair> chunks = this.chunks.stream().map(ChunkCoordIntPair::new).collect(Collectors.toSet());
+        TaskManager.IMP.task(() ->
+                {
+                    try {
+                        // cast is required for invokeExact
+                        int unused = (int) relight.invokeExact(world.getChunkProvider().getLightEngine(),
+                                (Set<?>) new HashSet<>(chunks), // explicit cast to make invokeExact work
+                                (Consumer<?>) coord -> {}, // no callback
+                                nothingIntConsumer
+                        );
+                    } catch (Throwable throwable) {
+                        LOGGER.error("Error occurred on relighting", throwable);
+                    }
+                }
+        );
+        if (Settings.IMP.LIGHTING.DELAY_PACKET_SENDING) {
+            for (long key : this.chunks) {
+                int x = ChunkCoordIntPair.getX(key);
+                int z = ChunkCoordIntPair.getZ(key);
+                ChunkHolder<?> chunk = (ChunkHolder<?>) queue.getOrCreateChunk(x, z);
+                IChunkGet toSend = chunk.getOrCreateGet();
+                TaskManager.IMP.async(() -> {
+                    Fawe.imp().getPlatformAdapter().sendChunk(toSend, -1, false);
+                });
+            }
+        }
+    }
+
+    @Override
+    public void clear() {
+
+    }
+
+    @Override
+    public void removeLighting() {
+
+    }
+
+    @Override
+    public void fixBlockLighting() {
+
+    }
+
+    @Override
+    public void fixSkyLighting() {
+
+    }
```

Comment:
The above methods should probably just refer to the fixLightingSafe method for sake of API-consistency.